Create weather data/iotools page in User's Guide (#1754)
pvlib/pvlib-python: docs/sphinx/source/user_guide/weather_data.rst

.. _weatherdata:

Weather data
============

Simulating the performance of a PV system requires irradiance and meteorological data
as the inputs to a PV system model.  Weather datasets are available

> from many sources and in many formats.  The :py:mod:`pvlib.iotools` module

Reviewer: Could we make a hyperlink to the iotools module for those who do not know what it is?

contains functions to easily retrieve and import such datasets in a standardized
form that is convenient to use with the rest of pvlib.  

The primary focus of :py:mod:`pvlib.iotools` is time series solar resource

> data like the datasets from PVGIS and the NSRDB, but it also provides

Reviewer: ```suggestion
data like the irradiance datasets from PVGIS and the NSRDB, but it also provides
```

> functionality for other types of data useful for certain aspects of PV modeling
> (e.g. precipitation data from :py:func:`~pvlib.iotools.get_acis_prism`

Reviewer: ```suggestion
functionality for other types of data useful for certain aspects of PV modeling.
For example, precipitation data is available from :py:func:`~pvlib.iotools.get_acis_prism`
```

for soiling modeling, and horizon profiles from :py:func:`~pvlib.iotools.get_pvgis_horizon`
for horizon shade modeling).

For a complete list of functions related to retrieving and importing weather
data, see :ref:`iotools`.

Types of weather data sources
-----------------------------

Weather data can be grouped into a few fundamental categories.  Which
type is most useful depends on the application.  Here we provide a high-level
overview of different types of weather data, and when you might want to use
them.

1. **Ground station measurements**:
   From in-situ monitoring equipment. If properly maintained and
   quality-controlled, these are the highest quality
   source of weather information. However, the coverage depends on
   a weather station having been set up in advance for the location and
   times of interest. Some ground station networks like the BSRN and SURFRAD

>    make their measurement data publicly available.

Reviewer: ```suggestion
   make their measurement data publicly available. An global overview of ground
   stations is available at https://SolarStations.org.
```

   
   Data from public ground station measurement networks are useful if you
   want accurate, high-resolution data but have flexibility around the
   specific measurement location.

2. **Satellite data**: 
   These sources process satellite imagery (typically from geostationary
   satellites) to identify and classify clouds, and combine this with solar

>    irradiance models to produce irradiance estimates. The quality is
>    generally much higher than NWP, but still not as good as a well-maintained

Reviewer: ```suggestion
   irradiance models and aerosol data to produce irradiance estimates. The
   quality is generally much higher than NWP, but still not as good as a well-maintained
```

   weather station. They have high spatial and temporal resolution
   corresponding to the source satellite imagery, and are generally
   optimised to estimate solar irradiance for PV applications. Free sources

>    such as PVGIS are available, and commerical sources such as SolarAnywhere,
>    Solcast and Solargis provide paid options though often have free trials.

Reviewer: ```suggestion
   such as PVGIS, NSRDB, and CAMS are available, and commerical sources such
   as SolarAnywhere, Solcast, and Solargis provide paid options though often
   have free trials.
```

>    Satellite data is useful when suitable ground station measurements are

Reviewer: ```suggestion
   Satellite data is useful when suitable ground measurements are
```

   not available for the location and/or times of interest.

3. **Numerical Weather Prediction (NWP)**:
   These are mathematical simulations of weather systems.
   The data quality is much lower than that of measurements and
   satellite data, owing in part to coarser spatial and temporal
   resolution, as well as many models not being optimised for solar
   irradiance for PV applications. On the plus side, these models typically
   have worldwide coverage, with some regional models (e.g. HRRR) sacrifice
   global coverage for somewhat higher spatial and temporal resolution.
   Various forecast (e.g. GFS, ECMWF, ICON) and reanalysis sources (ERA5,
   MERRA2) exist.
   
   NWP datasets are primarily useful for parts of the world not covered

>    by satellite-based datasets (e.g. the poles) or if extremely long time
>    ranges are needed.

Reviewer: ```suggestion
   by satellite-based datasets (e.g. polar regions) or if extremely long time
   ranges are needed.
```

For a more detailed comparison of the weather datasets available through

> pvlib, see [1]_.

Reviewer: Can we move this citation up to the intro (I doubt most folks would read this far)

> :py:mod:`pvlib.iotools` usage

Reviewer: ```suggestion
Usage
```

returning a :py:class:`pandas.DataFrame` containing the hourly weather data
and a python dict with information about the dataset.

Most :py:mod:`pvlib.iotools` functions work with time series datasets.
In that case, the returned ``df`` DataFrame has a datetime index, localized
to the appropriate time zone where possible.  Make sure to understand each
dataset's timestamping convention (e.g. center versus end of interval), as
pvlib will use these timestamps for solar position calculations.

The content of the metadata dictionary varies for each function/dataset.

Data retrieval
**************

Several :py:mod:`pvlib.iotools` functions access the internet to fetch data from
online web APIs.  For example, :py:func:`~pvlib.iotools.get_pvgis_hourly`
downloads data from PVGIS's webservers and returns it as a python variable.
Functions that retrieve data from the internet are named ``get_``, followed
by the name of the data source: :py:func:`~pvlib.iotools.get_bsrn`,
:py:func:`~pvlib.iotools.get_psm3`, :py:func:`~pvlib.iotools.get_pvgis_tmy`,
and so on.

For satellite/reanalysis datasets, the location is specified by latitude and
longitude in decimal degrees:

.. code-block:: python

>     lat, lon = 33.75, -84.39  # Atlanta, Georgia, United States
>     df, metadata = pvlib.iotools.get_psm3(lat, lon, map_variables=True, ...)

Reviewer: ```suggestion
    latitude, longitude = 33.75, -84.39  # Atlanta, Georgia, United States
    df, metadata = pvlib.iotools.get_psm3(latitude, longitude, map_variables=True, ...)
```

.. _weatherdata:

> Importing weather data
> ======================

Reviewer: ```suggestion
Weather data
=========
```

Reviewer: I think it's more than just "importing", it's also obtaining/retrieving weather data.

> Many PV modeling applications require irradiance and meteorological datasets
> as the inputs to a PV system model.  These weather datasets are available
> from many sources and in many formats.  The :py:mod:`pvlib.iotools` module
> contains functions to retrieve and import these datasets in a form
> that is convenient to use with the rest of pvlib.
> 

Reviewer: ```suggestion
Simulating the performance of a PV system requires irradiance and meteorological data
as the inputs to a PV system model.  Weather datasets are available
from many sources and in many formats.  The :py:mod:`pvlib.iotools` module
contains functions to easily retrieve and import such datasets in a standardized
form that is convenient to use with the rest of pvlib.

```
